osidb/models.py

```diff
@@ -1952,6 +1952,83 @@ def bzsync(self, *args, bz_api_key=None, **kwargs):
         self.flaw.save(*args, bz_api_key=bz_api_key, **kwargs)
 
 
+class FlawAcknowledgmentManager(ACLMixinManager, TrackingMixinManager):
+    """flaw acknowledgment manager"""
+
+    @staticmethod
+    def create_flawacknowledgment(
+        flaw, name, affiliation, from_upstream, **extra_fields
+    ):
+        """return a new flaw acknowledgment or update an existing flaw acknowledgment without saving"""
+        try:
+            flawacknowledgment = FlawAcknowledgment.objects.get(
+                flaw=flaw,
+                name=name,
+                affiliation=affiliation,
+                from_upstream=from_upstream,
```

Comment:
I am thinking whether we should add `from_upstream` to the `get` query or rather just always update it the same way as the `extra_fields` two lines below.

I am thinking about a situation when there is already an acknowledgment `flawX`, `nameY`, `affiliationZ`, `from_upstream==False` and someone will attempt to create another one  `flawX`, `nameY`, `affiliationZ`, `from_upstream==True` (same but different `from_upstream`). Then `get` will fail with `ObjectDoesNotExist` and the `except` block will fail violating the unique constrain.

Reply:
Agreed. This will allow for an easy switch/fix of the `from_upstream` bool. Alternatives might be more "correct", but much more complicated, comparatively. Fixed in an amended commit.